clang/runtime/dpct-rt/include/memory.hpp

```diff
@@ -977,141 +990,201 @@ async_dpct_memcpy(pitched_data to, sycl::id<3> to_pos, pitched_data from,
                                size_t y,
                                sycl::queue &q = get_default_queue()) {
   sycl::event::wait(detail::dpct_memset(q, ptr, pitch, val, x, y));
 }
 
-/// Sets \p value to the 2D memory region pointed by \p ptr in \p q. \p x and
-/// \p y specify the setted 2D memory size. \p pitch is the bytes in linear
-/// dimension, including padding bytes. The return of the function does NOT
-/// guarantee the memset operation is completed.
-///
-/// \param ptr Pointer to the device memory region.
-/// \param pitch Bytes in linear dimension, including padding bytes.
-/// \param value Value to be set.
-/// \param x The setted memory size in linear dimension.
-/// \param y The setted memory size in second dimension.
-/// \param q The queue in which the operation is done.
-/// \returns no return value.
+/**
+ * @brief Sets D8 \p val to the pitched 2D memory region pointed by \p ptr in \p q
+ * asynchronously.
+ * @param [in] ptr Pointer to the virtual device memory.
+ * @param [in] pitch The pitch size by number of elements, including padding.
+ * @param [in] val The value to be set.
+ * @param [in] x The width of memory region by number of elements.
+ * @param [in] y The height of memory region by number of elements.
+ * @param [in] q The queue in which the operation is done.
+ */
 static inline void async_dpct_memset(void *ptr, size_t pitch, int val, size_t x,
                                      size_t y,
                                      sycl::queue &q = get_default_queue()) {
   detail::dpct_memset<unsigned char>(q, ptr, pitch, val, x, y);
 }
 
-static inline void async_dpct_memset_d16(void *ptr, size_t pitch, unsigned short val, size_t x,
-                                     size_t y,
-                                     sycl::queue &q = get_default_queue()) {
+/**
+ * @brief Sets D16 \p val to the pitched 2D memory region pointed by \p ptr in \p q
+ * asynchronously.
+ * @param [in] ptr Pointer to the virtual device memory.
+ * @param [in] pitch The pitch size by number of elements, including padding.
+ * @param [in] val The value to be set.
+ * @param [in] x The width of memory region by number of elements.
+ * @param [in] y The height of memory region by number of elements.
+ * @param [in] q The queue in which the operation is done.
+ */
+static inline void async_dpct_memset_d16(void *ptr, size_t pitch,
+                                         unsigned short val, size_t x, size_t y,
+                                         sycl::queue &q = get_default_queue()) {
   detail::dpct_memset(q, ptr, pitch, val, x, y);
 }
 
-static inline void async_dpct_memset_d32(void *ptr, size_t pitch, unsigned int val, size_t x,
-                                     size_t y,
-                                     sycl::queue &q = get_default_queue()) {
+/**
+ * @brief Sets D32 \p val to the pitched 2D memory region pointed by \p ptr in \p q
```

Comment:
will end user know D32 here? If not need plan what is D32 or (32 bit data) instead